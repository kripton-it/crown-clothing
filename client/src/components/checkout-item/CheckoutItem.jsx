import React from "react";

import {
  CheckoutItemContainer,
  ImageContainer,
  ItemName,
  ItemQuantityContainer,
  ItemQuantityArrow,
  ItemQuantityValue,
  ItemPrice,
  RemoveButton
} from "./CheckoutItemStyles";

const CheckoutItem = ({
  cartItem,
  removeItem,
  addItemToCart,
  decreaseAmount
}) => {
  const { imageUrl, name, price, quantity } = cartItem;

  return (
    <CheckoutItemContainer>
      <ImageContainer>
        <img src={imageUrl} alt={name} />
      </ImageContainer>
      <ItemName>{name}</ItemName>
      <ItemQuantityContainer>
        <ItemQuantityArrow id="arrow-left" onClick={() => decreaseAmount(cartItem)}>
          &#10094;
        </ItemQuantityArrow>
        <ItemQuantityValue>{quantity}</ItemQuantityValue>
        <ItemQuantityArrow id="arrow-right" onClick={() => addItemToCart(cartItem)}>
          &#10095;
        </ItemQuantityArrow>
      </ItemQuantityContainer>
      <ItemPrice>{price}</ItemPrice>
      <RemoveButton className onClick={() => removeItem(cartItem)}>
        &#10005;
      </RemoveButton>
    </CheckoutItemContainer>
  );
};

export default CheckoutItem;
