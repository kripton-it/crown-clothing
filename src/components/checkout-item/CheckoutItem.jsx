import React from "react";

import { connect } from "react-redux";

import { removeItemFromCartAction } from "../../redux/cart/cart-actions";

import "./CheckoutItem.styles.scss";

const CheckoutItem = ({ cartItem, removeItem }) => {
  const { imageUrl, name, price, quantity, id } = cartItem;

  return (
    <div className="checkout-item">
      <div className="image-container">
        <img src={imageUrl} alt={name} />
      </div>
      <span className="name">{name}</span>
      <span className="quantity">{quantity}</span>
      <span className="price">{price}</span>
      <div className="remove-button" onClick={() => removeItem(id)}>
        &#10005;
      </div>
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  removeItem: id => dispatch(removeItemFromCartAction(id))
});

export default connect(
  null,
  mapDispatchToProps
)(CheckoutItem);
