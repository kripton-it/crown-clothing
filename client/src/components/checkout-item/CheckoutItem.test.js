import React from "react";
import { shallow } from "enzyme";

import CheckoutItem from "./CheckoutItem";

describe("CheckoutItem component", () => {
  let wrapper;
  const mockRemoveItem = jest.fn();
  const mockAddItemToCart = jest.fn();
  const mockDecreaseAmount = jest.fn();
  const mockCartItem = {
    imageUrl: "abc.com",
    price: 100,
    name: "hello",
    quantity: 5
  };

  beforeEach(() => {
    const mockProps = {
      cartItem: mockCartItem,
      removeItem: mockRemoveItem,
      addItemToCart: mockAddItemToCart,
      decreaseAmount: mockDecreaseAmount
    };

    wrapper = shallow(<CheckoutItem {...mockProps} />);
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("should render info:", () => {
    it("img", () => {
      const image = wrapper.find("ImageContainer img");
      expect(image.prop("src")).toBe("abc.com");
      expect(image.prop("alt")).toBe("hello");
    });

    it("name", () => {
      const itemName = wrapper.find("ItemName");
      expect(itemName.text()).toBe("hello");
    });

    it("quantity", () => {
      const itemQuantity = wrapper.find("ItemQuantityValue");
      expect(parseInt(itemQuantity.text())).toBe(5);
    });

    it("price", () => {
      const itemPrice = wrapper.find("ItemPrice");
      expect(parseInt(itemPrice.text())).toBe(100);
    });
  });

  describe("should call methods:", () => {
    it("should call removeItem when remove button is clicked", () => {
      wrapper.find("RemoveButton").simulate("click");
      expect(mockRemoveItem).toHaveBeenCalled();
    });

    it("should call decreaseAmount when left arrow is clicked", () => {
      wrapper.find("#arrow-left").simulate("click");

      expect(mockDecreaseAmount).toHaveBeenCalled();
    });

    it("should call addItemToCart when right arrow is clicked", () => {
      wrapper.find("#arrow-right").simulate("click");

      expect(mockAddItemToCart).toHaveBeenCalled();
    });
  });
});
