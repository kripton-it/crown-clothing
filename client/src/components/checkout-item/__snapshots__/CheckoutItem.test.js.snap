// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component should render 1`] = `ShallowWrapper {}`;
